File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        long ans = 0;
        while(x != 0){
            int lastDigit = x%10;
            x /= 10;
            ans = ans * 10 + lastDigit;
        }
        if(ans >= Integer.MIN_VALUE && ans <= Integer.MAX_VALUE) return (int)ans;
        return 0;
    }
}

File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int num = 0;
        while(x) {
            if (INT_MIN/10 > num || num > INT_MAX/10) return 0;
            num = num*10 + x%10;
            x /=10;            
        }
        return num;
    }
};